lib/OpenQA/Schema/Result/JobModules.pm

```diff
@@ -235,4 +235,35 @@ sub save_results {
     path($self->job->result_dir, 'details-' . $self->name . '.json')->spurt(encode_json($results));
 }
 
+# Collect text output into JSON. save_results() is called too early for this.
+sub finalize_results {
+    my ($self) = @_;
+
+    my $dir = $self->job->result_dir();
+    return unless $dir;
+    my $file = path($dir, "details-" . $self->name . ".json");
+    return unless -e $file;
+    my $results;
+
+    try {
+        $results = decode_json($file->slurp);
+    }
+    catch {
+        return;
```

Comment:
This `return;` will *not* make it leave the `finalize_results` function but only the catch block (which is actually a nested function). For that you could skip the catch block completely and use `return undef unless defined $results;` instead. Anyways: It seems inconsistent that this function lets exceptions from e.g. `slurp` and `spurt` functions invoked later pass but not exceptions from `decode_json`/`slurp` here invoked here. It would make sense to report the same type of errors consistently at least within one function.

Reply:
I copied the `try ... catch` block from `JobModules::results()`. I guess I should simply wrap the whole method in an `eval` block instead.

Reply:
You can also use `eval` and to return more easily in the error case.

However, I'm still wondering about the inconsistent error handling. Note that if this task was a Minion job (as suggested in the other comment) and not executed in response to an HTTP request I would simply let it die at this point.